test/acceptance/tests.py

```diff
@@ -794,16 +812,20 @@ def verify_submission_has_peer_grade(self, learner, max_attempts=5):
             learner: the learner whose grade will be checked
             max_attempts: the maximum number of times an additional peer grading should be done
         """
+        def peer_grade_exists():
+            self.staff_area_page.visit()
+            self.staff_area_page.show_learner(learner)
+            return "Peer Assessments for This Learner" in self.staff_area_page.learner_report_sections
+
         count = 0
-        while not self.peer_asmnt_page.is_complete and count < (max_attempts + 1):
+        while not peer_grade_exists() and count < max_attempts:
```

Comment:
"self.peer_asmnt_page.is_complete" is the wrong thing to check here. It verifies that the learner has submitted required peer assessments, not that someone has assessed them.

Previously this code checked "self.grade_page.wait_for_page().score is not None", which worked well for the case of staff override because the learner would get a score as soon as peer assessments were complete (staff would later override that score). However, @efischer19 changed the check in a previous PR to re-use this method for the case of full staff grading (in which the learner does NOT get a grade immediately when peer grading is complete).

What this method really needs to do is verify that peer assessment of this learner is complete. This is tricky because there is no LMS indication of this in the full grading case (or at least, due to TNL-3957, it is not clear what that indication should be). So instead I relied on using staff tools to see if peer assessments exist for the learner.